rustls-mbedcrypto-provider/src/agreement.rs

```diff
@@ -8,6 +8,7 @@
 use mbedtls::pk::{EcGroupId, ECDSA_MAX_LEN};
 
 /// An ECDH key agreement algorithm.
+#[derive(Debug)]
```

Comment:
You might want to think about whether you really want to derive Debug for this struct. While it doesn't contain any secrets, knowing these values might have value for an attacker.

Reply:
I think Francisco should probably look at this change. I'm out of my depth both for some of the Rust constructions and for reviewing crypto code.

Reply:
Make sense. I can actually avoid this line, since it's not necssary to debug print them (There is no clear use case to print them).

Reply:
removed

Reply:
In the same sense, is https://github.com/fortanix/rustls-mbedtls-provider/pull/65/files#diff-508887ea2da6ea0a0f89bc3212a612ca92878ca711ed5f8599604cdbc919a719R55 strictly necessary?

Reply:
Got it. This is another reason to not include `rng_provider` in the trait. Debugging the supported Key Exchange group is harmless (a TLS server obviously advertises this), but not the RNG used by the implementation.

Reply:
On the very least, we could not debug out the `rng_provider` field

Reply:
Got your idea. I will remove them.

Reply:
done